packages/dds/tree/src/treeFactory.ts

```diff
@@ -179,3 +190,24 @@ export function configuredSharedTree(
 
 	return makeSharedObjectKind<ITree>(sharedObjectOptions);
 }
+
+export function resolveOptions(options: SharedTreeOptions): SharedTreeOptionsInternal {
+	const internal: SharedTreeOptionsInternal = {
+		...resolveSharedBranchesOptions(options.enableSharedBranches),
+	};
+	copyProperty(options, "forest", internal);
```

Comment:
Why select particular options instead of just copying all options via ... operator and overriding the sharedBranchesOptions?

Reply:
Because that would include members that don't belong in the output struct. Not a huge problem, but very confusing when debugging.